internal/kgateway/metrics/collection_test.go

```diff
@@ -124,85 +87,69 @@ func TestCollectionResources(t *testing.T) {
-
-				if _, exists := resourceValues[*metric.Label[2].Value]; !exists {
-					resourceValues[*metric.Label[2].Value] = make(map[string]map[string]float64)
-				}
-
-				if _, exists := resourceValues[*metric.Label[2].Value][*metric.Label[1].Value]; !exists {
-					resourceValues[*metric.Label[2].Value][*metric.Label[1].Value] = make(map[string]float64)
-				}
-
-				resourceValues[*metric.Label[2].Value][*metric.Label[1].Value][*metric.Label[3].Value] = metric.Gauge.GetValue()
-			}
-
-			assert.Equal(t, float64(5), resourceValues["default"]["test"]["route"])
-			assert.Equal(t, float64(3), resourceValues["kube-system"]["test"]["gateway"])
-		}
+	expectedLabels := [][]*metricLabel{
+		{
+			{name: "collection", value: "test-collection"},
+			{name: "name", value: "test"},
+			{name: "namespace", value: "default"},
+			{name: "resource", value: "route"},
+		},
+		{
+			{name: "collection", value: "test-collection"},
+			{name: "name", value: "test"},
+			{name: "namespace", value: "kube-system"},
+			{name: "resource", value: "gateway"},
+		},
 	}
 
-	assert.True(t, found, "kgateway_collection_resources metric not found")
+	currentMetrics := mustGatherMetrics(t)
+
+	currentMetrics.assertMetricsLabels("kgateway_collection_resources", expectedLabels)
+	currentMetrics.assertMetricGaugeValues("kgateway_collection_resources", []float64{5, 3})
 
 	// Test IncResources.
 	m.IncResources(CollectionResourcesLabels{Namespace: "default", Name: "test", Resource: "route"})
 
-	metricFamilies, err = metrics.Registry.Gather()
-	require.NoError(t, err)
-
-	for _, mf := range metricFamilies {
-		if *mf.Name == "kgateway_collection_resources" {
-			for _, metric := range mf.Metric {
-				if len(metric.Label) > 0 && *metric.Label[0].Value == "default" {
-					assert.Equal(t, float64(6), metric.Gauge.GetValue())
-				}
-			}
-		}
-	}
+	currentMetrics = mustGatherMetrics(t)
+	currentMetrics.assertMetricsLabels("kgateway_collection_resources", expectedLabels)
+	currentMetrics.assertMetricGaugeValues("kgateway_collection_resources", []float64{6, 3})
 
 	// Test DecResources.
 	m.DecResources(CollectionResourcesLabels{Namespace: "default", Name: "test", Resource: "route"})
 
-	metricFamilies, err = metrics.Registry.Gather()
-	require.NoError(t, err)
+	currentMetrics = mustGatherMetrics(t)
+	currentMetrics.assertMetricsLabels("kgateway_collection_resources", expectedLabels)
+	currentMetrics.assertMetricGaugeValues("kgateway_collection_resources", []float64{5, 3})
 
+	// Test ResetResources.
+	m.ResetResources("test")
+
+	metricFamilies, err := metrics.Registry.Gather()
+	require.NoError(t, err)
 	for _, mf := range metricFamilies {
 		if *mf.Name == "kgateway_collection_resources" {
 			for _, metric := range mf.Metric {
+				// Todo: namespace:default is label[2]. Index wrong?
 				if len(metric.Label) > 0 && *metric.Label[0].Value == "default" {
 					assert.Equal(t, float64(5), metric.Gauge.GetValue())
 				}
 			}
 		}
 	}
 
+	// TODO: why call twice?
```

Comment:
Don't call the time above this L125. That was added in your commit, it should not be there.